src/writing-policies/rust.md

```diff
@@ -1,25 +1,647 @@
+    let tc = Testcase {
+        name: String::from("Pod creation with valid name"),
+        fixture_file: String::from(request_file),
+        expected_validation_result: true,
+        settings,
+    };
+
+    let res = tc.eval(validate).unwrap();
+    // NOTE 1
+    assert!(
+        res.mutated_object.is_some(),
+        "Expected accepted object to be mutated",
+    );
+
+    // NOTE 2
+    let final_pod =
+        serde_json::from_str::<apicore::Pod>(res.mutated_object.unwrap().as_str()).unwrap();
+    let final_annotations = final_pod.metadata.annotations.unwrap();
+    assert_eq!(
+        final_annotations.get_key_value("kubewarden.policy.demo/inspected"),
+        Some((
+            &String::from("kubewarden.policy.demo/inspected"),
+            &String::from("true")
+        )),
+    );
+
+    Ok(())
+}
+```
+
+Compared to the initial test, we made only two changes:
+
+  1. Change the `assert!` statement to ensure the request is still accepted,
+    but it also includes a mutated object
+  2. Created a `Pod` instance starting from the mutated object that is part of
+    the response. Assert the mutated Pod object contains the right
+    `metadata.annotations`.
+
+We can run the tests again, this time all of them will pass:
+
+```shell
+$ cargo test
+   Compiling demo v0.1.0 (/home/flavio/hacking/kubernetes/kubewarden/demo)
+    Finished test [unoptimized + debuginfo] target(s) in 2.61s
+     Running target/debug/deps/demo-24670dd6a538fd72
+
+running 5 tests
+test settings::tests::reject_settings_without_a_list_of_invalid_names ... ok
+test settings::tests::accept_settings_with_a_list_of_invalid_names ... ok
+test tests::accept_request_with_non_pod_resource ... ok
+test tests::reject_pod_with_invalid_name ... ok
+test tests::accept_pod_with_valid_name ... ok
+
+test result: ok. 5 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
+```
+
+As you can see the creation of a mutation policy is pretty straightforward.
+
+## Building the policy
+
+So far we have built the policy using as a compilation target the same operating
+system and architecture of our development machine.
+
+It's not time to build the policy as a WebAssembly binary, also known as `.wasm`
+file.
+
+This can be done with a simple command:
+
+```shell
+make build
+```
+
+This command will build the code in release mode, with WebAssembly as
+compilation target.
+
+The build will produce the following file:
+
+```shell
+$ file target/wasm32-unknown-unknown/release/demo.wasm
+target/wasm32-unknown-unknown/release/demo.wasm: WebAssembly (wasm) binary module version 0x1 (MVP)
+```
 
-WebAssembly is a first-class citizen in the Rust world, that means
-many of the tools and crates should work out of the box.
+## Distributing the policy
 
-Of course, Rust programs are still bound to the limitations of the WASI
-specification.
+This topic is covered inside of the ["distributing policies"](/distributing-policies.html)
```

Comment:
Is quoting the link itself intended?

Reply:
Yep, that's the title of the docs section. I would leave that if you don't mind

Reply:
Yes, was just double checking that it was what you wanted